cmd/govim/internal/fswatcher/os_darwin.go

```diff
@@ -13,17 +14,19 @@ import (
 	"gopkg.in/tomb.v2"
 )
 
-const fRemoved = fsevents.ItemRemoved | fsevents.ItemRenamed
-const fChanged = fsevents.ItemModified | fsevents.ItemChangeOwner
-const fCreated = fsevents.ItemCreated
+const (
+	fRemoved = fsevents.ItemRemoved | fsevents.ItemRenamed
+	fChanged = fsevents.ItemModified | fsevents.ItemChangeOwner
+	fCreated = fsevents.ItemCreated
+)
 
 type fswatcher struct {
 	eventCh chan Event
 	es      *fsevents.EventStream
 }
 
-func New(gomodpath string, tomb *tomb.Tomb) (*FSWatcher, error) {
-	dirpath := filepath.Dir(gomodpath)
+func New(root string, filter watchFilterFn, logf logFn, tomb *tomb.Tomb) (*FSWatcher, error) {
+	dirpath := filepath.Dir(root)
```

Comment:
We seem to calculate `filepath.Dir` here, but from what I can tell `root` will only ever be a directory? 

And why do we end up watching the parent directory of `root`? A comment here would be useful too.

Reply:
Good catch, it's a leftover from before when the go.mod path was passed in. We shouldn't watch the `root` parent, just the root. Fixed.